packages/did/src/Did.utils.ts

```diff
@@ -104,37 +131,50 @@ export function parseDidUri(
   didUri: DidUri | DidResourceUri
 ): IDidParsingResult {
   let matches = FULL_KILT_DID_REGEX.exec(didUri)?.groups
-  if (matches && matches.identifier) {
-    const version = matches.version
-      ? parseInt(matches.version, 10)
+  if (matches) {
+    const { version: versionString, fragment } = matches
+    const address = matches.address as KiltAddress
+    const version = versionString
+      ? parseInt(versionString, 10)
       : FULL_DID_LATEST_VERSION
     return {
-      did: getKiltDidFromIdentifier(matches.identifier, 'full', version),
+      did: getKiltDidFromIdentifier(address, 'full', version),
       version,
       type: 'full',
-      identifier: matches.identifier,
-      fragment: matches.fragment?.substring(1),
+      identifier: address,
+      address,
+      fragment: fragment === '#' ? undefined : (fragment as UriFragment),
     }
   }
 
   // If it fails to parse full DID, try with light DID
   matches = LIGHT_KILT_DID_REGEX.exec(didUri)?.groups
-  if (matches && matches.identifier && matches.auth_key_type) {
-    const version = matches.version ? parseInt(matches.version, 10) : 1
-    const lightDidIdentifier = matches.auth_key_type.concat(matches.identifier)
-    const encodedDetails = matches.encoded_details?.substring(1)
+  if (matches) {
+    const {
+      authKeyType,
+      version: versionString,
+      encodedDetails,
+      fragment,
+    } = matches
+    const address = matches.address as KiltAddress
+    const version = versionString
+      ? parseInt(versionString, 10)
+      : LIGHT_DID_LATEST_VERSION
+    const identifier = `${authKeyType}${address}` as DidIdentifier
```

Comment:
You might be able to type the variable here instead of using type assertion. Would give you the benefit of actual type checking.

Reply:
Since TS doesn’t have insight into the regular expression, it doesn’t know possible values of `authKeyType` and complains:

>TS2322: Type '`${string}4${string}`' is not assignable to type 'DidIdentifier'.

Reply:
ah, too bad